ambari-server/src/main/java/org/apache/ambari/server/controller/internal/ClusterResourceProvider.java

```diff
@@ -533,7 +533,8 @@ private RequestStatusResponse processBlueprintCreate(Map<String, Object> propert
 
     ProvisionClusterRequest createClusterRequest;
     try {
-      createClusterRequest = topologyRequestFactory.createProvisionClusterRequest(properties, securityConfiguration);
+      createClusterRequest =
+        topologyRequestFactory.createProvisionClusterRequest(rawRequestBody, properties, securityConfiguration);
```

Comment:
Maybe at this point only the "filtered" document should be passed around, meaning that the passwords have already been removed from the document.